docs/explanation/integration.md

```diff
@@ -0,0 +1,134 @@
+# Integration test workflow
+
+The integration test is composed of the follwing steps.
+
+1. [Plan the workflow](#plan-the-workflow)
+2. [Build the artifacts](#build-the-artifacts)
+3. [Run the integration test](#run-the-integration-test)
+
+This documentation goes over each steps, covering the internal details of how
+they work.
+
+## Plan the workflow
+
+The [plan workflow](../../internal/plan/action.yml) refers to the 
+[plan action](../../src/plan.ts). It is used to plan the build for the following
+resources if detected:
+
+* Charm
+* Rock
+* Docker
+* Charm Resources
+
+The resources are shared between workflows using the local filesystem or the
+GitHub container registry, depending on the user specified `upload-image` input.
+
+When the workflow is executed from a fork, the resources are shared via the
+filesystem.
+
+### Charm
+
+The workflow will automatically look for any `charmcraft.yaml` file or the 
+`metadata.yaml` file within the source repository root by using the following
```

Comment:
It only looks for the `charmcraft.yaml` file, the `metadata.yaml` file is only used to extract charm property (if the `metadata.yaml` file exists).

Reply:
Updated to:
```
The workflow will automatically look for any `charmcraft.yaml` file within the
source repository root by using the following glob syntax: `**/charmcraft.yaml`.
`metadata.yaml` file is also used to extract any required charm property if it
is found. Any `charmcraft.yaml` files within the `tests/` directory are ignored.
```

Thank you!